### PR TITLE
Update for scalability

### PR DESCRIPTION
**Pull request by ggeorgakoudis**
- Synchronize to create launcher after worker pods are running
- Use IPs instead of hostnames to bypass DNS

### CODE REVIEW

**Comment by ggeorgakoudis** [comment]
Probably need to fix the mpi_job_controller_test.go, leaving that for after getting some feedback

**Comment by alculquicondor** [comment]
AFAIK, static MPI applications don't pay attention to changes in the hostfile.

Do you have stats on how much this is saving?

File: examples/v2beta1/pi/pi.yaml [hunk 1]
```diff
@@ -35,6 +30,12 @@ spec:
           containers:
           - image: mpioperator/mpi-pi
             name: mpi-worker
+            lifecycle:
+              postStart:
+            # exec command below makes sure ssh service on port 22 is up and running -- checking localhost but should be
```
Comment (alculquicondor) [inline examples/v2beta1/pi/pi.yaml:35]: But what does this prevent?

Were you meant to use a readinessProbe instead? I set up one for the intel scenario https://github.com/kubeflow/mpi-operator/blob/master/examples/v2beta1/pi/pi-intel.yaml

File: examples/v2beta1/pi/pi.yaml [hunk 2]
```diff
@@ -18,12 +18,7 @@ spec:
             name: mpi-launcher
             securityContext:
               runAsUser: 1000
-            command:
-            - mpirun
-            args:
-            - -n
-            - "2"
-            - /home/mpiuser/pi
+            command: [ "/bin/bash", "-cx", "mpirun -n 128 --oversubscribe /home/mpiuser/pi"]
```
Comment (alculquicondor) [inline examples/v2beta1/pi/pi.yaml:21]: This change doesn't look relevant to the intention of the PR. Can you revert?

**Comment by ggeorgakoudis** [comment]
Awaiting LLNL review for CLA agreement, currently code is under LLNL copyright
